src/cpp/src/llm_pipeline_npu.cpp

```diff
@@ -0,0 +1,305 @@
+                                          partial_shape[3].get_length()});
+        }
+        new_shapes.emplace(input_name, new_shape);
+    }
+    model->reshape(new_shapes);
+}
+
+void fill_tensor(ov::Tensor tensor, int64_t fill_val) {
+    int64_t* tensor_data = tensor.data<int64_t>();
+    std::fill(tensor_data, tensor_data + tensor.get_size(), fill_val);
+}
+
+void copy_with_left_offset(const ov::Tensor& orig, ov::Tensor& padded) {
+    const auto orig_size = orig.get_size();
+    const auto padded_size = padded.get_size();
+    const auto kLeftOffset = padded_size - orig_size;
+    int64_t* orig_data = orig.data<int64_t>();
+    int64_t* padded_data = padded.data<int64_t>();
+    std::copy(orig_data, orig_data + orig_size, padded_data + kLeftOffset);
+}
+
+} // anonymous namespace
+
+namespace ov {
+namespace genai {
+
+NPULLMPipelineImpl::NPULLMPipelineImpl(
+    const std::filesystem::path& path,
+    const ov::genai::Tokenizer& tokenizer,
+    const ov::AnyMap& config
+) : LLMPipelineImplBase(tokenizer,
+                        utils::from_config_json_if_exists(path)) {
+    /* NB: NPU-friendly LLM pipeline consists of two models,
+       first to process the input prompt (prefill), second to use in generation loop (kvcache)
+
+       Initialization assumes multiple steps:
+       1) Read the template model and clone it for the further use as kvcache and prefill models
+       2) Expose KV-cache input and output layers for the kvcache model
+       3) Reshape both models to static shape
+       4) Add slices to KV-cache inputs for kvcache model, this will make input and output KV-cache
+          layers to have the same shape and allow outputs writes directly to inputs for the next iteration.
+       5) Compile both models for NPUW
+       6) Initialize input tensors for kvcache and prefill models
+    */
+
+    ov::Core core;
+    // (1) Read the template model - this will be kvcache
+    auto kvcache_model = core.read_model(path / "openvino_model.xml");
+    // (2) Expose KV-cache input and output layers
+    ov::pass::StatefulToStateless().run_on_model(kvcache_model);
+    // (3) Clone the model - this will be prefill
+    auto prefill_model = kvcache_model->clone();
+    prefill_model->set_friendly_name(kvcache_model->get_friendly_name() + "_prefill");
+    // (4) Reshape both models to static shape
+    // FIXME: There must be better logic than just hardcoded values
+    m_kvcache_desc = KVCacheDesc { 1024u, 0u };
+    const uint32_t max_prompt_size = m_kvcache_desc.total_size;
+    const uint32_t max_kvcache_size = m_kvcache_desc.total_size;
+    reshape_to_static(prefill_model, max_prompt_size, max_kvcache_size);
+    reshape_to_static(kvcache_model, 1u, max_kvcache_size);
+    // (5) Add slices to kvcache model
+    kvcache_model = add_slices_to_kvcache_inputs(kvcache_model);
+    // (6) Compiled both models for NPUW
+    std::map<std::string, std::string> cfg = { {"NPU_USE_NPUW", "YES"},
```

Comment:
Must gone from there, consider extracting this information from `plugin_config` as temporal solution. Ideally it will be just `USE_NPUW=YES`

Reply:
Done